### PR TITLE
JSAP-13: deposit account details backend

### PR DESCRIPTION
**Pull request by x566chen**

### CODE REVIEW

File: backend/src/models/DatabaseActions.js [hunk 1]
```diff
@@ -32,7 +32,15 @@ module.exports = class DatabaseActions {
       throw error;
     }
   }
-
+  async getAccountDetails(userId, depositId) {
+      try {
```
Comment (matecserven) [inline backend/src/models/DatabaseActions.js]:
```suggestion
    try {
```

check your commit hook again

File: backend/src/tests/depositRoute.test.js [hunk 2]
```diff
@@ -73,3 +73,53 @@ describe('Get all accounts Endpoint', () => {
     ]);
   });
 });
+
+describe('Get accounts Endpoint we want', () => {
+  it('should return all transaction', async () => {
+    const res = await request(app)
+      .get('/deposit/3/2');
+    // expect(res.statusCode).toEqual(200);
```
Comment (matecserven) [inline backend/src/tests/depositRoute.test.js]:
```suggestion
    expect(res.statusCode).toEqual(200);
```

File: backend/src/routes/depositRoute.js [hunk 3]
```diff
@@ -41,5 +41,13 @@ depositRoute.get('/:userId', async (req, res) => {
     res.status(500).send('Something went wrong, please try again later.');
   }
 });
+depositRoute.get('/:userId/:depositId', async(req, res)=>{
```
Comment (matecserven) [inline backend/src/routes/depositRoute.js]:
```suggestion
depositRoute.get('/:depositId', async(req, res)=>{
```
This is no longer the way the service works, user id comes from the token

File: backend/src/tests/depositRoute.test.js [hunk 4]
```diff
@@ -73,3 +73,53 @@ describe('Get all accounts Endpoint', () => {
     ]);
   });
 });
+
+describe('Get accounts Endpoint we want', () => {
+  it('should return all transaction', async () => {
+    const res = await request(app)
```
Comment (matecserven) [inline backend/src/tests/depositRoute.test.js]: you need to mock the database action

File: backend/db/migrations/1574749257606_transferDetails.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+    "up": 'CREATE TABLE IF NOT EXISTS transferDetails ('
+    + 'transferId INT PRIMARY KEY AUTO_INCREMENT, '
+    + 'transferAmount INT,'
+    + 'fromUserId INT,'
+    + 'fromAccount INT,'
```
Comment (matecserven) [inline backend/db/migrations/1574749257606_transferDetails.js]:
```suggestion
    + 'fromDepositId INT,'
```

File: backend/db/migrations/1574749257606_transferDetails.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    + 'toUserId INT,'
+    + 'toAccount INT,'
```
Comment (matecserven) [inline backend/db/migrations/1574749257606_transferDetails.js]:
```suggestion
    + 'toDepositId INT,'
```

File: backend/src/tests/depositRoute.test.js [hunk 7]
```diff
@@ -73,3 +73,53 @@ describe('Get all accounts Endpoint', () => {
...
+        transferId: 5,
+        transferAmount: 150,
+        fromUserId: 4,
+        fromDepositId: 3,
+        toUserId: 2,
+        toDepositId: 2,
+        status: 'pending',
+        timeOfTransfer: 12.9,
+      },
+      {
```
Comment (matecserven) [inline backend/src/tests/depositRoute.test.js]: How do you get this result? you are searching for the id = 2, you got 3 and 6 as result, how is this possible? Also you still haven't mocked the database action and you are not using authentication token, this test should not work. Rebase to master and check how the tests should work

Comment (x566chen) [reply backend/src/tests/depositRoute.test.js]: sry, it is the result from last time, I mean when I use /:userId/:depositId...and i forgot to modify results in test file..my fault. really sry about that.

File: backend/src/models/DatabaseActions.js [hunk 8]
```diff
@@ -33,6 +33,16 @@ module.exports = class DatabaseActions {
     }
   }
 
+  async getAccountDetails(depositId) {
+    try {
+      const findAccountDetailsQuery = 'SELECT * FROM transferDetails WHERE fromDepositId = ? OR toDepositId = ?;';
```
Comment (matecserven) [inline backend/src/models/DatabaseActions.js]: Before you send the details back, it should be checked if the deposit accounts are definitely the user's or not, you can probably implement this in the query, user ID comes from the token through the request

File: backend/db/migrations/1574750872770_transferDetails.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  up: 'INSERT INTO transferDetails(transferAmount, fromUserId, fromDepositId, toUserId, toDepositId, status) '
+    + 'VALUES '
+    + '( 150, 123, 1, 456, 4, "pending"), '
+    + '( 150, 123, 3,456, 6, "pending"), '
+    + '( 150, 456, 5, 789, 8, "pending"), '
+    + '( 90, 789, 9, 456, 5, "pending"), '
+    + '( 150, 456, 5, 456, 5, "pending"), '
+    + '( 150, 456, 5, 123, 2, "pending"), '
+    + '( 100, 456, 4, 456, 5, "done");',
```
Comment (matecserven) [inline backend/db/migrations/1574750872770_transferDetails.js]: please add these as well
    + '( 99, 123, 1, 456, 5, "pending"), '
    + '( 50, 123, 1, 789, 8, "pending"), '
    + '( 99, 456, 4, 123, 1, "pending"), '
    + '( 50, 789, 9, 123, 2, "pending"), '

File: backend/db/migrations/1574750872770_transferDetails.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    + '( 150, 123, 3,456, 6, "pending"), '
+    + '( 150, 456, 5, 789, 8, "pending"), '
+    + '( 90, 789, 9, 456, 5, "pending"), '
+    + '( 150, 456, 5, 456, 5, "pending"), '
+    + '( 150, 456, 5, 123, 2, "pending"), '
+    + '( 100, 456, 4, 456, 5, "done"),'
+    + '( 99, 123, 1, 456, 5, "pending"), '
+    + '( 50, 123, 1, 789, 8, "pending"), '
+    + '( 99, 456, 4, 123, 1, "pending"), '
+    + '( 50, 789, 9, 123, 2, "pending"), ',
```
Comment (matecserven) [inline backend/db/migrations/1574750872770_transferDetails.js]:
```suggestion
    + '( 50, 789, 9, 123, 2, "pending");',
```

**Pull request merged**